### ISSUE TITLE
Rewrite "package:" imports in Dart files in build output

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#854]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [munificent](https://github.com/munificent)**
_Originally opened as dart-lang/sdk#16649_

----

If you run pub build in debug, you'll get Dart files in the build output. There will likely be other ways in the future of getting Dart files in your output. If they contain &quot;package:&quot; imports, those will fail unless the Dart file happens to be in the top level of the build output where we put the one copy of &quot;packages&quot;.

For all other cases, there is no adjacent &quot;packages&quot; directory. To fix that, we want a built-in transformer in pub that rewrites those imports to relative ones that walk up to the main &quot;packages&quot; directory.

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Marked this as being blocked by dart-lang/pub#846.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Added [Duplicate](../labels/Duplicate) label.*
*Marked as being merged into dart-lang/pub#799.*
